WalletWasabi.Standard/CoinJoin/Coordinator/Rounds/CoordinatorRound.cs

```diff
@@ -1345,6 +1346,40 @@ private async Task<(bool accept, string rejectReason)> TestMempoolAcceptWithTran
 			return await RpcClient.TestMempoolAcceptAsync(coinsToTest, fakeOutputCount: outputCount, FeePerInputs, FeePerOutputs).ConfigureAwait(false);
 		}
 
+		public void QueueAlices(params Guid[] ids)
+		{
+			using (RoundSynchronizerLock.Lock())
+			{
+				if ((Phase != RoundPhase.InputRegistration && Phase != RoundPhase.ConnectionConfirmation) || Status != CoordinatorRoundStatus.Running)
+				{
+					throw new InvalidOperationException("Queuing Alice is only allowed in InputRegistration and ConnectionConfirmation phases.");
+				}
+				foreach (var id in ids)
+				{
+					var alices = Alices.Where(x => x.UniqueId == id);
+					foreach (var alice in alices)
+					{
+						QueuedAlices.Add(alice);
+					}
```

Comment:
This assumes that a re-queued alice gets the same id, right?

The nested loops are a bit smelly to me. What about 

goal Quue unique alices by id
```cs
ids.Select(id => Alices.Where(a => a.UniqueId == id))
    .Distinct()
    .Select(unique => QueuedAlices.Add(unique));
```

I'm betting yet cleaner is possible 🤷🏻‍♂️.